src/OSPSuite.Core/Serialization/Xml/ColumnXmlSerializer.cs

```diff
@@ -12,14 +13,23 @@ public override void PerformMapping()
          Map(x => x.LloqColumn);
          Map(x => x.Name);
          Map(x => x.Unit);
+         Map(x => x.Dimension);
       }
 
       protected override void TypedDeserialize(Column objectToDeserialize, XElement outputToDeserialize, SerializationContext context)
       {
          base.TypedDeserialize(objectToDeserialize, outputToDeserialize, context);
-         
+
          if (objectToDeserialize.ErrorStdDev == Constants.STD_DEV_GEOMETRIC)
+         {
             objectToDeserialize.Unit = new UnitDescription();
+            objectToDeserialize.Dimension = Constants.Dimension.NO_DIMENSION;
+         }
+
+         if (!objectToDeserialize.Unit.ColumnName.IsNullOrEmpty())
+         {
+            objectToDeserialize.Dimension = null;
```

Comment:
and why can this dimensio be NOT null already? It does not look like we have a lot of confidence in what is being serialized. Can we talk about this live?

Reply:
I agree with you @msevestre. I have just made some changes that should make the process more reliable. So, before there was a problem when saving the dimension in case of geometric error. Now the dimension is properly set when serializing so we do not need any check when deserializing. I know @georgeDaskalakis was fixing some different errors so I would like to discuss these changes with him when he returns so we are sure we did not overlooked anything and that is why I am adding this comment so I know where to look after.